more_itertools/more.py

```diff
@@ -180,18 +180,19 @@ def last(iterable, default=_marker):
     raise ``ValueError``.
     """
     try:
-        try:
-            # Try to access the last item directly
+        if isinstance(iterable, Sequence):
             return iterable[-1]
-        except (TypeError, AttributeError, KeyError):
-            # If not slice-able, iterate entirely using length-1 deque
-            return deque(iterable, maxlen=1)[0]
-    except IndexError as e:  # If the iterable was empty
+        # Work around https://bugs.python.org/issue38525
+        elif hasattr(iterable, '__reversed__') and (hexversion != 0x030800F0):
```

Comment:
same here, I would try/except rather than checking `hasattr`

Reply:
I would have done `isinstance(iterable, Reversible)`, but `Reversible` isn't in Python 3.5. But is [effectively](https://github.com/python/cpython/blob/3.8/Lib/_collections_abc.py#L305) a check for a `__reversed__` method.